lib/shared/testing/types.ts

```diff
@@ -0,0 +1,25 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview Utility functions used only in test code, that allow tests to
+ * take advantage of TypeScript's flow-control-based type checking.
+ */
+
+/** Throws if the given condition is false. */
+export function assert(condition: boolean): asserts condition {
+  if (!condition) {
+    throw new TypeError("Assertion failure");
+  }
+}
+
+/** Throws if the given value is null or undefined; returns it otherwise. */
+export function nonNullish<T>(value: T): NonNullable<T> {
+  assert(value != null);
+  // https://github.com/typescript-eslint/typescript-eslint/issues/3310
+  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion, @typescript-eslint/no-unnecessary-type-assertion
+  return value!;
+}
```

Comment:
What do you think of the proposal at https://github.com/typescript-eslint/typescript-eslint/issues/3310#issuecomment-826008027

Reply:
Might have return-only generics issues, but it's probably not a big deal, so I'll switch to using it.